Tests/FluentAssertions.Equivalency.Specs/DataRowSpecs.cs

```diff
@@ -335,6 +335,47 @@ public void When_data_row_data_does_not_match_but_the_column_is_excluded_then_eq
             dataRow1.Should().BeEquivalentTo(dataRow2, config => config.ExcludingColumn(dataSet2.TypedDataTable1.DateTimeColumn));
         }
 
+        [Fact]
+        public void Data_row_is_not_equivalent_to_another_type()
+        {
+            // Arrange
+            var table = new DataTable();
+            var dataRow = table.NewRow();
+            var subject = new
+            {
+                DataRow = "foobar"
+            };
+
+            var expected = new
+            {
+                DataRow = dataRow
+            };
+
+            // Act
+            Action act = () => subject.Should().BeEquivalentTo(expected);
+
+            // Assert
+            act.Should().Throw<XunitException>()
+                .WithMessage("Expected*System.Data.DataRow*found System.String*");
+        }
+
+        [Fact]
+        public void When_subject_of_type_DataRowCollection_is_expected_and_is_not_it_fails()
```

Comment:
What case are you exercising here?
Asking since the `subject` has a property `DataRowCollection` but I don't imagine that being used in the test.

Reply:
Because with other solutions it does not get into the branch 
DataRowCollectionEquivalencyStep.cs,L#13-18